test/tests/functional/pbs_qrun.py

```diff
@@ -92,3 +92,48 @@ def test_invalid_host_val(self):
         msg = "Server is not up"
         self.assertTrue(self.server.isUp(), msg)
         self.logger.info("As expected server is up and running")
+
+    def test_qrun_hangs(self):
+        """
+        This test submit 500 jobs with differnt equivalence class,
+        turn of scheduling and qrun job to
+        verify whether qrun hangs.
+        """
+        node = self.mom.shortname
+        err_msg = 'qrun: Not enough free nodes available'
+        joblist = []
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'False'})
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {'resources_available.ncpus': 1}, id=node)
+        for walltime in range(1, 501):
+            j = Job(TEST_USER)
+            a = {'Resource_List.walltime': walltime}
+            j.set_attributes(a)
+            jid = self.server.submit(j)
+            joblist.append(jid)
+        self.logger.info("Submitted 500 jobs with different walltime")
+
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'True'})
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'False'})
+        jid = joblist[10]  # it scheduled a few jobs
```

Comment:
Why not qrun the last job id? That way you'll have the least chance of running into a race condition.

Reply:
Yes sure. Then I don't need to save all jobid also in a list.
Will remove the list too.